### PR TITLE
impl(compute): handle optional sample arg

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12767)

### CODE REVIEW

File: google/cloud/compute/samples/disk_samples.cc [hunk 1]
```diff
@@ -87,7 +87,8 @@ void CreateEmptyDisk(std::vector<std::string> const& argv) {
   }
   //! [END compute_disk_create_empty_disk]
   (argv.at(0), argv.at(1), argv.at(2), argv.at(3),
-   SplitLabelsString(argv.at(4)));
+   ((argv.size() == 5) ? SplitLabelsString(argv.at(4))
```
Comment (devbww) [inline google/cloud/compute/samples/disk_samples.cc:90]: I think now I'd change the "throw Usage" condition to `if (argv.size() != 4 && argv.size() != 5)`.

**Review by devbww — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12767)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @devbww)

___
*[`google/cloud/compute/samples/disk_samples.cc` line 90 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12767#gh-1341643422:-NfX0NhoCnskb5mf_CTJ:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/8beadec99b361a6971fd2292d4b430f8ab21084f/google/cloud/compute/samples/disk_samples.cc#L90)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

I think now I'd change the "throw Usage" condition to `if (argv.size() != 4 && argv.size() != 5)`.
</blockquote></details>

Done.

File: google/cloud/compute/samples/disk_samples.cc [hunk 2]
```diff
@@ -87,7 +87,8 @@ void CreateEmptyDisk(std::vector<std::string> const& argv) {
...
+                       : std::vector<std::pair<std::string, std::string>>{}));
```
Comment (devbww) [inline google/cloud/compute/samples/disk_samples.cc:91]: Aside: You might also consider that `std::vector<std::pair<std::string, std::string>>` smells like `std::[multi]map<std::string, std::string>`, and particularly seeing as the latter appears to match `Disk.labels`.

**Review by devbww — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12767)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @devbww)

___
*[`google/cloud/compute/samples/disk_samples.cc` line 91 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12767#gh-1341652485:-NfX4moV0YY6wiZcyg-O:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/7f54b735852883f24e88ce280321ab3cb55417fe/google/cloud/compute/samples/disk_samples.cc#L91)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Aside: You might also consider that `std::vector<std::pair<std::string, std::string>>` smells like `std::[multi]map<std::string, std::string>`, and particularly seeing as the latter appears to match `Disk.labels`.
</blockquote></details>

Done.

File: google/cloud/compute/samples/disk_samples.cc [hunk 3]
```diff
@@ -26,9 +26,8 @@
 
 namespace {
 
-std::vector<std::pair<std::string, std::string>> SplitLabelsString(
-    std::string const& arg) {
-  std::vector<std::pair<std::string, std::string>> labels;
+std::map<std::string, std::string> SplitLabelsString(std::string const& arg) {
```
Comment (devbww) [inline google/cloud/compute/samples/disk_samples.cc:29]: `#include <map>`

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12767)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @devbww)

___
*[`google/cloud/compute/samples/disk_samples.cc` line 29 at r3](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12767#gh-1341668422:-NfX9oj44beUf59sWZeW:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/e438399c49745e129df8d64a1381d256defb478f/google/cloud/compute/samples/disk_samples.cc#L29)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

`#include <map>`
</blockquote></details>

Done.

**Pull request merged**